Classes/CloudFlareAPI.cs

```diff
@@ -21,221 +21,357 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
+
 using System;
-using System.Text;
+using System.Collections.Generic;
 using System.IO;
 using System.Net;
+using System.Net.Sockets;
+using System.Text;
 using System.Web.Script.Serialization;
+using CloudFlareDDNS.Classes.JsonObjects.Cloudflare;
 
 namespace CloudFlareDDNS
 {
     /// <summary>
     /// Provides functions for interfacing with the CloudFlare REST API
     /// </summary>
-    class CloudFlareAPI
+    internal class CloudFlareAPI
     {
-
-
+        JavaScriptSerializer jss = new JavaScriptSerializer();
         /// <summary>
         /// Logic to update records
+        /// And return changes
         /// </summary>
-        public void updateRecords(JsonResponse fetchedRecords)
+        public List<Result> updateRecords(Get_dns_records_response fetchedRecords)
         {
+            //List for the Updated IPs
+            List<Result> return_updated_list = new List<Result>();
+
             if (fetchedRecords == null) //Dont attempt updates if the fetch failed
-                return;
+                return return_updated_list;
 
             int up_to_date = 0, skipped = 0, failed = 0, updated = 0, ignored = 0;
             string[] selectedHosts = Program.settingsManager.getSetting("SelectedHosts").ToString().Split(';');
 
-            for (int i = 0; i < fetchedRecords.response.recs.count; i++)
+            foreach (Result r in fetchedRecords.result)
             {
                 //Skip over MX and CNAME records
                 //TODO: Dont skip them :)
-                if (fetchedRecords.response.recs.objs[i].type != "A")
+                bool NeedIp = false;

```

Comment:
This can be merged into a switch, with default case.
case "A": // fall through.
case: "AAAA":
   NeedIp = true;
default:
   NeedIp = false;